src/periodicnotes/periodicnoteentry.ts

```diff
@@ -0,0 +1,86 @@
+import { App, TAbstractFile, TFile, Notice } from "obsidian";
+import type { Moment } from "moment";
+import type { ClippedNoteEntry } from "../clippednoteentry";
+import { SectionPosition } from "./sectionposition";
+import { PrependWriter } from "./prependwriter";
+import { AppendWriter } from "./appendwriter";
+
+export abstract class PeriodicNoteEntry {
+  protected app: App;
+  protected notice: string;
+  protected openFileOnWrite: boolean;
+  protected sectionPosition: SectionPosition;
+
+  protected abstract getPeriodicNote(
+    moment: Moment,
+    allNotes: Record<string, TFile>
+  ): TFile;
+  protected abstract hasPeriodicNoteEnabled(): boolean;
+  protected abstract getAllNotes(): Record<string, TFile>;
+  protected abstract waitForNoteCreation(moment: Moment): Promise<TFile>;
+
+  constructor(
+    app: App,
+    openFileOnWrite: boolean,
+    sectionPosition: SectionPosition
+  ) {
+    this.app = app;
+    this.openFileOnWrite = openFileOnWrite;
+    this.sectionPosition = sectionPosition;
+  }
+
+  async writeToPeriodicNote(noteEntry: ClippedNoteEntry, heading: string) {
+    if (!this.hasPeriodicNoteEnabled()) {
+      new Notice(this.notice);
+      return;
+    }
+
+    let note = await this.getNote();
+
+    this.handleWrite(
+      note.path,
+      await noteEntry.formattedEntry(),
+      this.sectionPosition,
+      heading
+    );
+  }
+
+  protected async getNote() {
+    let now = window.moment();
```

Comment:
**question:** Is there a reason you're getting at `moment` from `window`? Could it be imported instead?

Reply:
as a follow-up, if `window` is the proper place, then maybe replace it with `globalThis`

```suggestion
    const now = globalThis.moment();
```